beacon_node/eth2-libp2p/src/behaviour/handler/mod.rs

```diff
@@ -77,15 +89,25 @@ impl<TSpec: EthSpec> ProtocolsHandler for BehaviourHandler<TSpec> {
         out: <Self::OutboundProtocol as OutboundUpgrade<NegotiatedSubstream>>::Output,
         info: Self::OutboundOpenInfo,
     ) {
+        if self.keep_alive < KeepAlive::Yes {
+            // shutting down, reject all outbounds
```

Comment:
I think you have a check for this in the RPC handler also. I think it's fine to allow streams once they have been estalished and let them finish. 
In the RPC handler we will throw an error and drop the stream.. for the other behaviours, lets just let them go. I can't think of any reason why this would be an issue at the moment. 

We will prevent further requests for streams but if they are in the process of being negotiated and get negotiated, lets leave them be.